testing/tests/headless.test.js

```diff
@@ -0,0 +1,28 @@
+require("expect-puppeteer");
+
+function testURL(pathname = "/") {
+  return "http://localhost:5000" + pathname;
+}
+
+describe("Basic viewing of functional pages", () => {
+  it("open the temporary home page", async () => {
+    await page.goto(testURL("/"));
```

Comment:
I assume `page` is injected into the `it` function's namespace somehow by `expect-puppeteer`?

Reply:
Does that bother you too?!
I've come to accept it but I've never liked that about `jest`. I wish it was:
```javascript
const {describe, it, expect, test} from 'jest';
```

But yes, that `page` becomes magically available. To be honest I don't know how it works but I can imagine there's a `beforeEach` that creates a new `browser` instance and then creates a page from it and which recycled after the test so that cookies and localStorage and stuff don't persist between tests.